### PR TITLE
 For #12930 - Unregister FXA observers to prevent memory leaks 

### LINKED ISSUE #12930: Ensure cleanup of the tabs presenter to prevent memory leaks
As a followup to https://github.com/mozilla-mobile/fenix/issues/27374 where it was seen that it is possible to leak an entire layout through `feature.syncedtabs.presenter.DefaultPresenter` and `FxaAccountManager`.

### PR DESCRIPTION
**Pull request by Mugurell**
No leaks in the scenario of opening and closing tabs tray on Fenix:

https://user-images.githubusercontent.com/11428869/195330861-e44619d0-3dcf-4941-957c-8d044c85feca.mp4

### Pull Request checklist

- [x] **Quality**: This PR builds and passes detekt/ktlint checks (A pre-push hook is recommended)
- [x] **Tests**: This PR includes thorough tests or an explanation of why it does not
- [x] **Changelog**: This PR includes [a changelog entry](https://github.com/mozilla-mobile/android-components/blob/main/docs/changelog.md) or does not need one
- [x] **Accessibility**: The code in this PR follows [accessibility best practices](https://github.com/mozilla-mobile/shared-docs/blob/main/android/accessibility_guide.md) or does not include any user facing features

### After merge
- **Milestone**: Make sure issues closed by this pull request are added to the [milestone](https://github.com/mozilla-mobile/android-components/milestones) of the version currently in development.
- **Breaking Changes**: If this is a breaking change, please push a draft PR on [Reference Browser](https://github.com/mozilla-mobile/reference-browser) to address the breaking issues.

### GitHub Automation
Fixes #12930

### CODE REVIEW

File: components/service/firefox-accounts/src/main/java/mozilla/components/service/fxa/manager/FxaAccountManager.kt [hunk 1]
```diff
@@ -487,6 +493,14 @@ open class FxaAccountManager(
         syncStatusObserverRegistry.register(observer, owner, autoPause)
     }
 
+    /**
+     * Unregister a [SyncStatusObserver] from being informed about "sync lifecycle" events.
+     * The method is safe to call even if the provided observer was not registered before.
+     */
+    fun unregisterForSyncEvents(observer: SyncStatusObserver) {
```
Comment (jonalmeida) [inline components/service/firefox-accounts/src/main/java/mozilla/components/service/fxa/manager/FxaAccountManager.kt:500]: Didn't realize we didn't have an unregister before. 😄

**Review by jonalmeida — APPROVED** [review]
Thanks for the quick fix! 🔥

File: components/service/firefox-accounts/src/main/java/mozilla/components/service/fxa/manager/FxaAccountManager.kt (hunk 1 above)
Comment (Mugurell) [reply components/service/firefox-accounts/src/main/java/mozilla/components/service/fxa/manager/FxaAccountManager.kt:500]: This probably means there could've been other leaks also but hopefully after this they'll be no more.

Comment (jonalmeida) [reply components/service/firefox-accounts/src/main/java/mozilla/components/service/fxa/manager/FxaAccountManager.kt:500]: Yeah, and I see others on the AccountFragment in Fenix too, but none of the sync events observer.

**Pull request merged**